Makefile.pre.in

```diff
@@ -1994,6 +1994,24 @@ testuniversal: all
 	$(RUNSHARED) /usr/libexec/oah/translate \
 		./$(BUILDPYTHON) -E -m test -j 0 -u all $(TESTOPTS)
 
+# Run the test suite on the iOS simulator. Must be run on a macOS machine with
+# a full Xcode install that has an iPhone SE (3rd edition) simulator available.
+XCFOLDER=$(srcdir)/build/iOSTestbed.$(MULTIARCH).$(shell date +%s)
+XCRESULT=$(XCFOLDER)/$(MULTIARCH).xcresult
+.PHONY: testios
+testios: install
```

Comment:
I haven't started implementing this yet, but for Android I was thinking of running the tests as a subcommand of the [same script that does the build](https://github.com/mhsmith/cpython/blob/android/Android/android.py), which is itself based on the [existing WASI script](https://github.com/python/cpython/blob/main/Tools/wasm/wasi.py). In other words, the top-level script run by the buildbot would be something like:
```
./android.py build [args...]
./android.py test [args...]
```

Maybe this would be easier than incorporating the test script into the Makefile, especially since it already includes a block of Python code.

Reply:
@ned-deily I've removed the `install` dependency, and added some extra safety checks around the `testios` target to check whether the target is in a state where it can be executed. I've also added some documentation clarifying the role of the `install` target as a "not really 'installing' per se, but still required" step.

@mhsmith For the single-platform iOS case, I'm not sure if a separate script is needed. The modifications here are enough to build and test for a single platform, and they aren't overly complex. There's likely a case for a script to do the full "merged" build (i.e., configuring multiple platform builds, then merging the final XCframework), but that's a subject for a future PR. 

FWIW, the existing logic in [Python-apple-support](https://github.com/beeware/Python-Apple-support/blob/main/Makefile#L635) demonstrates that you *can* build an XCframework with Makefile logic... but  it may well be easier to use a WASI-style build management script, especially for the process of merging the simulator builds into a single fat framework, which currently relies on some *very* dark Makefile magic.